src/main/java/io/cdap/plugin/gcp/dataplex/sink/DataplexBatchSink.java

```diff
@@ -556,6 +606,121 @@ private void emitMetricsForStorageBucket(boolean succeeded, BatchSinkContext con
+    }
+
+    if (entityBean != null && entityBean.getSchema().getUserManaged() == false) {
+      collector.addFailure("Entity already exists, but the schema is not user-managed.", null);
+    }
+  }
+
+  /**
+   * Configures metadata update in Dataplex if the schema is user managed
+   *
+   * @param credentials
+   * @param assetFullPath
+   * @param storageSystem
+   * @param schema
+   * @throws IOException
+   */
+  private void configureMetadataUpdateWithUserManagedSchema(
+    GoogleCredentials credentials, String assetFullPath, StorageSystem storageSystem, Schema schema
+  ) throws IOException {
+    String entityID = config.getTable().replaceAll("[^a-zA-Z0-9_]", "_");
+    try (MetadataServiceClient metadataServiceClient =
+           DataplexUtil.getMetadataServiceClient(credentials)) {
+      com.google.cloud.dataplex.v1.Schema dataplexSchema = DataplexUtil.getDataplexSchema(schema);
+      Entity.Builder entityBuilder = Entity.newBuilder()
+        .setId(entityID)
+        .setAsset(config.getAsset())
+        .setDataPath(assetFullPath)
+        .setType(Entity.Type.TABLE)
+        .setSystem(storageSystem)
+        .setSchema(dataplexSchema)
+        .setFormat(StorageFormat
+                     .newBuilder()
+                     .setMimeType(DataplexUtil.getStorageFormatForEntity(config.getFormatStr()))
+                     .build()
+        );
+
+      if (entityBean != null) {
+        try {
+          entityBean = metadataServiceClient.updateEntity(
+            UpdateEntityRequest.newBuilder()
+              .setEntity(entityBuilder
+                           .setName(entityBean.getName())
+                           .setEtag(entityBean.getEtag())
+                           .build()
+              )
+              .build()
+          );
+        } catch (ApiException e) {
+          throw new RuntimeException(
+            String.format("%s: %s", "There was a problem updating the entity for metadata updates.", e.getMessage()));
+        }
+      } else {
+        try {
+          String entityParent = "projects/" + config.tryGetProject() +
+            "/locations/" + config.getLocation() +
+            "/lakes/" + config.getLake() +
+            "/zones/" + config.getZone();
+          entityBean = metadataServiceClient.createEntity(
+            CreateEntityRequest.newBuilder()
+              .setParent(entityParent)
+              .setEntity(entityBuilder.build())
+              .build()
+          );
+        } catch (ApiException e) {
+          throw new RuntimeException(
+            String.format("%s: %s", "There was a problem creating the entity for metadata updates.", e.getMessage()));
+        }
+      }
+      try {
+        DataplexUtil.addPartitionInfo(entityBean, credentials,
+                                    asset.getResourceSpec().getName(), config.getTable(), config.getProject());
+      } catch (ApiException e) {
+        String errMessage = e.getMessage().substring(e.getMessage().length() - 14);
```

Comment:
why are we subtracting 14?

Reply:
Can you add a comment to clarify. 
Also, is it possible to figure out if it's an "already exists" error based on error code instead of parsing the message?

Reply:
The status code comes as `INVALID_ARGUMENT`. I thought that may be too generic, hence parsing the last 14 chars of the error message to make sure it's failing due to `already exists` error.

Reply:
I can just use http status code 400 instead.